### PR TITLE
Create email

### PR DESCRIPTION
**Pull request by tudorSV**

### CODE REVIEW

File: app/controllers/appointments_controller.rb [hunk 1]
```diff
@@ -9,9 +9,11 @@ def new
   def create
     @shop = Shop.find(params[:appointment][:shop_id])
     @appointment = Appointment.new(appointment_params.merge(user_id: @user.id,
-                                                             shop_id: @shop.id))
+                                                            shop_id: @shop.id))
     if @appointment.save
       flash[:success] = 'The appointment has been added'
+      AppointmentMailer.user_email(@appointment).deliver_now
```
Comment (sv-pdragos) [inline app/controllers/appointments_controller.rb]: use a better name for the email. one that suggests what the email is for. Don't forget to rename the template

File: app/controllers/appointments_controller.rb [hunk 2]
```diff
@@ -9,9 +9,11 @@ def new
...
+      AppointmentMailer.shop_email(@appointment).deliver_now
```
Comment (sv-pdragos) [inline app/controllers/appointments_controller.rb]: use a better name for the email. one that suggests what the email is for. Don't forget to rename the template

File: app/controllers/shops_controller.rb [hunk 3]
```diff
@@ -54,18 +54,21 @@ def index_appointment
...
   def change_status
+    @appointment = Appointment.find(params[:appointment])
     @company = Company.find(params[:company_id])
     authorize! :show, @company
     @shop = Shop.find(params[:shop_id])
     authorize! :show, @shop
     @appointment = Appointment.find(params[:appointment])
     if @appointment.status == 'Booked'
       @appointment.update(status: 'Ready for pickup')
+      AppointmentMailer.status_email(@appointment).deliver_now
```
Comment (sv-pdragos) [inline app/controllers/shops_controller.rb]: use a better name for the email. one that suggests what the email is for. Don't forget to rename the template

File: app/controllers/shops_controller.rb [hunk 4]
```diff
@@ -54,18 +54,21 @@ def index_appointment
...
       flash[:success] = 'The appointment has been updated to ready for pickup!'
       return redirect_to company_shop_index_appointment_path(@company, @shop)
     elsif @appointment.status == 'Ready for pickup'
       @appointment.update(status: 'Finished')
-      flash[:success] = 'The appointment has been completed!'
+      AppointmentMailer.thank_you_email(@appointment).deliver_now
```
Comment (sv-pdragos) [inline app/controllers/shops_controller.rb]: use a better name for the email. one that suggests what the email is for. Don't forget to rename the template

File: app/controllers/shops_controller.rb [hunk 5]
```diff
@@ -54,18 +54,21 @@ def index_appointment
...
+      flash[:success] = 'The appointment has been updated!'
```
Comment (sv-pdragos) [inline app/controllers/shops_controller.rb]: updated to what?

File: app/mailers/appointment_mailer.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class AppointmentMailer < ActionMailer::Base
+  default from: 'example@email.com'
```
Comment (sv-pdragos) [inline app/mailers/appointment_mailer.rb]: line not needed

File: app/mailers/appointment_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  def user_email(appointment)
+    @appointment = appointment
+    mail(to: @appointment.user.email, subject: 'Appointment created')
```
Comment (sv-pdragos) [inline app/mailers/appointment_mailer.rb]: You need a more descriptive subject

File: app/mailers/appointment_mailer.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def shop_email(appointment)
+    @appointment = appointment
+    mail(to: @appointment.shop.email, subject: 'Appointment created')
```
Comment (sv-pdragos) [inline app/mailers/appointment_mailer.rb]: You need a more descriptive subject

File: app/mailers/appointment_mailer.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def status_email(appointment)
+    @appointment = appointment
+    mail(to: @appointment.user.email, subject: 'Appointment update')
```
Comment (sv-pdragos) [inline app/mailers/appointment_mailer.rb]: You need a more descriptive subject

File: app/mailers/appointment_mailer.rb [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def thank_you_email(appointment)
+    @appointment = appointment
+    mail(to: @appointment.user.email, subject: 'Appointment has been completed')
```
Comment (sv-pdragos) [inline app/mailers/appointment_mailer.rb]: You need a more descriptive subject

File: app/models/appointment.rb [hunk 11]
```diff
@@ -3,8 +3,10 @@ class Appointment < ApplicationRecord
   belongs_to :user
 
   validates :date, presence: true
-  validates :item_number, presence: true
   validates :status, presence: true
+  validates :item_number, presence: true
+  validate :appointment_date
+  validate :available_date
```
Comment (sv-pdragos) [inline app/models/appointment.rb:9]: you duplicated this

File: app/models/appointment.rb [hunk 12]
```diff
@@ -3,8 +3,10 @@ class Appointment < ApplicationRecord
   belongs_to :user
 
   validates :date, presence: true
-  validates :item_number, presence: true
   validates :status, presence: true
+  validates :item_number, presence: true
+  validate :appointment_date
```
Comment (sv-pdragos) [inline app/models/appointment.rb:8]: you duplicated this

File: config/environments/development.rb [hunk 13]
```diff
@@ -58,4 +58,9 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  config.action_mailer.delivery_method = :letter_opener
+  config.action_mailer.perform_deliveries = true
+
+  config.action_mailer.delivery_method = :smtp
```
Comment (sv-pdragos) [inline config/environments/development.rb]: line not needed. in dev we use letter opener

File: config/initializers/assets.rb [hunk 14]
```diff
@@ -7,6 +7,7 @@
 # Rails.application.config.assets.paths << Emoji.images_path
 # Add Yarn node_modules folder to the asset load path.
 Rails.application.config.assets.paths << Rails.root.join('node_modules')
+Rails.application.config.assets.precompile += %w(email.css.scss)
```
Comment (sv-pdragos) [inline config/initializers/assets.rb]: you don''t need this

File: app/views/appointment_mailer/user_email.html.erb [hunk 15]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
```
Comment (sv-pdragos) [inline app/views/appointment_mailer/user_email.html.erb]: you only need the content here not the full structure

File: app/views/appointment_mailer/thank_you_email.html.erb [hunk 16]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
```
Comment (sv-pdragos) [inline app/views/appointment_mailer/thank_you_email.html.erb]: you only need the content here not the full structure

File: app/views/appointment_mailer/status_email.html.erb [hunk 17]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
```
Comment (sv-pdragos) [inline app/views/appointment_mailer/status_email.html.erb]: you only need the content here not the full structure

File: app/views/appointment_mailer/shop_email.html.erb [hunk 18]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
```
Comment (sv-pdragos) [inline app/views/appointment_mailer/shop_email.html.erb]: you only need the content here not the full structure

**Review by sv-pdragos — CHANGES_REQUESTED** [review]

File: app/controllers/appointments_controller.rb (hunk 1 above)
Comment (tudorSV) [reply app/controllers/appointments_controller.rb]: user_email has been changed to user_appointment_confirmation_email

File: app/controllers/appointments_controller.rb (hunk 2 above)
Comment (tudorSV) [reply app/controllers/appointments_controller.rb]: shop_email has been changed to shop_appointment_confirmation_email

File: app/controllers/shops_controller.rb (hunk 3 above)
Comment (tudorSV) [reply app/controllers/shops_controller.rb]: name has been changed to change_appointment_status_email

File: app/controllers/shops_controller.rb (hunk 4 above)
Comment (tudorSV) [reply app/controllers/shops_controller.rb]: name has been changed to user_thank_you_email

File: app/controllers/shops_controller.rb (hunk 5 above)
Comment (tudorSV) [reply app/controllers/shops_controller.rb]: success message has been changed

File: app/mailers/appointment_mailer.rb (hunk 7 above)
Comment (tudorSV) [reply app/mailers/appointment_mailer.rb]: subject has been changed

File: app/mailers/appointment_mailer.rb (hunk 8 above)
Comment (tudorSV) [reply app/mailers/appointment_mailer.rb]: subject has been changed

File: app/mailers/appointment_mailer.rb (hunk 9 above)
Comment (tudorSV) [reply app/mailers/appointment_mailer.rb]: subject has been changed

File: app/mailers/appointment_mailer.rb (hunk 10 above)
Comment (tudorSV) [reply app/mailers/appointment_mailer.rb]: subject has been changed

**Review by sv-pdragos — APPROVED** [review]

**Pull request merged**